src/redset.h

```diff
@@ -28,6 +28,27 @@ extern "C" {
 #define REDSET_COPY_XOR     (3)
 #define REDSET_COPY_RS      (4)
 
+/* names of user settable config parameters */
+#define REDSET_KEY_CONFIG_ENABLED   "ENABLED"
+#define REDSET_KEY_CONFIG_INTERVAL  "INTERVAL"
+#define REDSET_KEY_CONFIG_OUTPUT    "OUTPUT"
+#define REDSET_KEY_CONFIG_STORE     "STORE"
+#define REDSET_KEY_CONFIG_DIRECTORY "DIR"
+#define REDSET_KEY_CONFIG_TYPE      "TYPE"
+#define REDSET_KEY_CONFIG_SET_SIZE  "SETSIZE"
+#define REDSET_KEY_CONFIG_GROUP      "GROUP"
+#define REDSET_KEY_CONFIG_GROUPS     "GROUPS"
+#define REDSET_KEY_CONFIG_GROUP_ID   "GROUP"
+#define REDSET_KEY_CONFIG_GROUP_SIZE "RANKS"
+#define REDSET_KEY_CONFIG_GROUP_RANK "RANK"
+#define REDSET_KEY_CONFIG_MPI_BUF_SIZE "MPI_BUF_SIZE"
+#define REDSET_KEY_CONFIG_DEBUG "DEBUG"
+
+#define REDSET_KEY_COPY_XOR_RANKS "RANKS"
+#define REDSET_KEY_COPY_XOR_GROUP "GROUP"
+#define REDSET_KEY_COPY_XOR_GROUP_RANK  "RANK"
+#define REDSET_KEY_COPY_XOR_GROUP_RANKS "RANKS"
```

Comment:
Let's just expose the minimal set we support for config options here.

Reply:
These I think already existed since redset itself serializes things and then deserializes them at one point. The only ones actually configurable by a user are (https://github.com/ECP-VeloC/redset/pull/19/commits/c90464195b7877be432825689b81ae3c005e3297#diff-32c19e43a29cdb52ef5ede073a155824R61):

* `REDSET_KEY_CONFIG_DEBUG`
* `REDSET_KEY_CONFIG_SET_SIZE`
* `REDSET_KEY_CONFIG_MPI_BUF_SIZE`

Reply:
Hah, seeing this there was either a merge / rebase glitch or I forgot to remove the lines myself. These defines are from the various "*_serial.c" files and I moved them into the (global) header since some (`MPI_BUF_SIZE`) are user controllable and not used only in serializing / unserializing. The fact that both use `_CONFIG` in their names is unfortunate.

Reply:
It looks like a lot of these strings are still in the redset.h.  I'd like to only expose strings users care about in the main header.  Did you have a commit that moves them?

Reply:
Moved all except `REDSET_KEY_CONFIG_DEBUG`,  `REDSET_KEY_CONFIG_MPI_BUF_SIZE`, `REDSET_KEY_CONFIG_SET_SIZE` to redset_internal.h in 57acada.